util/util.go

```diff
@@ -295,6 +295,7 @@ func HasOwnerRef(ownerReferences []metav1.OwnerReference, ref metav1.OwnerRefere
 }
 
 // EnsureOwnerRef makes sure the slice contains the OwnerReference.
+// Note EnsureOwnerRef will update the version of the OwnerReference if it exists with a different version.
```

Comment:
Thank you very much for adding this comment!!

Can you please take a look at similar functions around here and add the same if it makes sense? (e.g. I was stumbling over RemoveOwnerRef)